### PR TITLE
Use model watcher

### PR DESCRIPTION
**Pull request by zsoltk**

### CODE REVIEW

**Comment by ShikaSD** [comment]
Still not quite sure if we want to enforce it
But PR looks ok

File: android/templates/snapshot/src/main/java/com/badoo/ribs/template/rib_with_view/foo_bar/FooBarView.kt [hunk 1]
```diff
@@ -37,11 +39,20 @@ class FooBarViewImpl private constructor(
...
         override fun invoke(deps: Nothing?): (ViewGroup) -> FooBarView = {
             FooBarViewImpl(
-                com.badoo.ribs.customisation.inflate(it, layoutRes)
+                inflate(it, layoutRes)
             )
         }
     }
 
-    override fun accept(vm: FooBarView.ViewModel) {
+    // feel free to check https://badoo.github.io/MVICore/extras/modelwatcher/ for usage examples
```
Comment (lukaville) [inline android/templates/snapshot/src/main/java/com/badoo/ribs/template/rib_with_view/foo_bar/FooBarView.kt]: do we need this comment? maybe we can put it in `modelWatcher` javadoc? currently, we should always remove it

File: android/templates/release-0.5/src/main/java/com/badoo/ribs/template/rib_with_view/foo_bar/FooBarView.kt [hunk 2]
```diff
@@ -37,11 +39,20 @@ class FooBarViewImpl private constructor(
...
         override fun invoke(deps: Nothing?): (ViewGroup) -> FooBarView = {
             FooBarViewImpl(
-                com.badoo.ribs.customisation.inflate(it, layoutRes)
+                inflate(it, layoutRes)
             )
         }
     }
 
-    override fun accept(vm: FooBarView.ViewModel) {
+    // feel free to check https://badoo.github.io/MVICore/extras/modelwatcher/ for usage examples
```
Comment (Nublo) [inline android/templates/release-0.5/src/main/java/com/badoo/ribs/template/rib_with_view/foo_bar/FooBarView.kt]: Should we force usage of ModelWatcher for customers of the Framework?
Maybe we need to mention it once with context of applying changes for separate fields of the model.

But I am not sure, that we need to force using it. You can have simple RIB with one textField and probably you don't want to use modelWatcher for this case. We are adding more and more complex stuff which can be repulsive for customers of the BadooRIBs.

Comment (zsoltk) [reply android/templates/release-0.5/src/main/java/com/badoo/ribs/template/rib_with_view/foo_bar/FooBarView.kt]: You are probably right, thanks. Can we make it not feel forced, just optional and encouraged?

**Pull request closed without merging**